BitwardenShared/UI/Auth/TwoFactorNotice/EmailAccess/EmailAccessProcessor.swift

```diff
@@ -0,0 +1,76 @@
+import Combine
+import SwiftUI
+
+// MARK: - EmailAccessProcessor
+
+/// The processor used to manage state and handle actions for the new device notice screen.
+///
+class EmailAccessProcessor: StateProcessor<EmailAccessState, EmailAccessAction, EmailAccessEffect> {
+    // MARK: Types
+
+    typealias Services = HasErrorReporter
+        & HasStateService
+
+    // MARK: Private Properties
+
+    /// The coordinator that handles navigation.
+    private let coordinator: AnyCoordinator<TwoFactorNoticeRoute, Void>
+
+    /// The services required by this processor.
+    private let services: Services
+
+    // MARK: Initialization
+
+    /// Creates a new `EmailAccessProcessor`.
+    ///
+    /// - Parameters:
+    ///   - coordinator: The coordinator that handles navigation.
+    ///   - services: The services required by this processor.
+    ///   - state: The initial state of the processor.
+    ///
+    init(
+        coordinator: AnyCoordinator<TwoFactorNoticeRoute, Void>,
+        services: Services,
+        state: EmailAccessState
+    ) {
+        self.coordinator = coordinator
+        self.services = services
+
+        super.init(state: state)
+    }
+
+    // MARK: Methods
+
+    override func perform(_ effect: EmailAccessEffect) async {
+        switch effect {
+        case .continueTapped:
+            await handleContinue()
+        }
+    }
+
+    override func receive(_ action: EmailAccessAction) {
+        switch action {
+        case let .canAccessEmailChanged(canAccess):
+            state.canAccessEmail = canAccess
+        }
+    }
+
+    // MARK: Private Methods
+
+    /// Checks the state of the UI when the user taps the continue button
+    /// and routes accordingly.
+    private func handleContinue() async {
+        do {
+            if state.canAccessEmail {
+                let displayState: TwoFactorNoticeDisplayState
+                displayState = state.allowDelay ? .canAccessEmail : .canAccessEmailPermanent
+                try await services.stateService.setTwoFactorNoticeDisplayState(state: displayState)
+                coordinator.navigate(to: .dismiss)
+            } else {
+                coordinator.navigate(to: .setUpTwoFactor(allowDelay: state.allowDelay))
+            }
+        } catch {
+            services.errorReporter.log(error: error)
```

Comment:
🤔 Shouldn't we display something to the user that states that something wrong happened? If not, the user may think the app got frozen or something like that if an error is thrown here but nothing is shown to the user.

Reply:
The only way something could go wrong here is if it failed to save to the state service, which shouldn't actually happen, in practice, I think? After all, that only fails if there's no active account, and you can't get here without an active account. I'm not sure what sort of message we should display here if that fails, though—"Please restart the app and try again"?